src/artemis/external_interaction/callbacks/rotation/zocalo_callback.py

```diff
@@ -0,0 +1,36 @@
+from __future__ import annotations
+
+from bluesky.callbacks import CallbackBase
+
+from artemis.external_interaction.callbacks.rotation.ispyb_callback import (
+    RotationISPyBHandlerCallback,
+)
+from artemis.external_interaction.exceptions import ISPyBDepositionNotMade
+from artemis.external_interaction.zocalo.zocalo_interaction import ZocaloInteractor
+from artemis.log import LOGGER
+
+
+class RotationZocaloHandlerCallback(CallbackBase):
+    def __init__(
+        self,
+        zocalo_environment: str,
+        ispyb_handler: RotationISPyBHandlerCallback,
+    ):
+        self.ispyb: RotationISPyBHandlerCallback = ispyb_handler
+        self.zocalo_interactor = ZocaloInteractor(zocalo_environment)
+
+    def start(self, doc: dict):
+        LOGGER.info("Zocalo handler received start document.")
+        if doc.get("subplan_name") == "rotation_scan_main":
+            self.run_uid = doc.get("uid")
+
+    def stop(self, doc: dict):
+        if doc.get("run_start") == self.run_uid:
+            LOGGER.info(
+                f"Zocalo handler received stop document, for run {doc.get('run_start')}."
+            )
+            if self.ispyb.ispyb_ids[0] is not None:
+                self.zocalo_interactor.run_start(self.ispyb.ispyb_ids[0])
+                self.zocalo_interactor.run_end(self.ispyb.ispyb_ids[0])
```

Comment:
Should: For grid scans we start this in `start`, why not here too?

Reply:
That's what I did first; however, start is before ispyb data is read, and so there is no deposition and no IDs yet. I thought about adding yet another layer to the plan to manage this, but since the result of this isn't time-sensitive, I thought it wasn't worth cluttering up the plan over.

Reply:
Can we make an issue to think about it maybe? I guess it doesn't matter if we're not doing parallel analysis but if we move to parallel analysis at some point it would be nice if it just worked

Reply:
I don't think it will ever be necessary, but we should discuss with scientists